raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -111,17 +111,17 @@
     private final int electionTimeoutMs;
     private final int electionJitterMs;
     private final int retryBackoffMs;
-    private final int requestTimeoutMs;
     private final long bootTimestamp;
     private final InetSocketAddress advertisedListener;
     private final NetworkChannel channel;
     private final ReplicatedLog log;
     private final QuorumState quorum;
     private final Random random;
     private final ConnectionCache connections;
+    private final RecordAppender recordAppender;
```

Comment:
Rather than having this as a field, could we construct a new one after each leader change? Then the epoch could be constant in the `RecordAppender` and we have an easy way to check that the state machine has seen the right events.